File: packages/sdk-php/src/EmailDomains.php
<?php

namespace Notifique;

/**
 * Email Domains — GET/POST /v1/email/domains, GET /v1/email/domains/:id, POST /v1/email/domains/:id/verify
 */
class EmailDomains
{
    private Notifique $client;

    public function __construct(Notifique $client)
    {
        $this->client = $client;
    }

    /** GET /v1/email/domains */
    public function list(): array
    {
        return $this->client->request('GET', '/email/domains', null);
    }

    /** POST /v1/email/domains */
    public function create(array $params): array
    {
        return $this->client->request('POST', '/email/domains', $params);
    }

    /** GET /v1/email/domains/:id */
    public function get(string $id): array
    {
        return $this->client->request('GET', '/email/domains/' . Notifique::encodePathSegment($id), null);
    }

    /** POST /v1/email/domains/:id/verify */
    public function verify(string $id): array
    {
        return $this->client->request('POST', '/email/domains/' . Notifique::encodePathSegment($id) . '/verify', null);
    }
}

File: packages/sdk-elixir/lib/notifique/email_domains.ex
defmodule Notifique.EmailDomains do
  @moduledoc """
  Email Domains — GET/POST /v1/email/domains, GET /v1/email/domains/:id, POST /v1/email/domains/:id/verify
  """

  def list(client, opts \\ []) do
    Notifique.request(client, :get, "/email/domains", nil, opts)
  end

  def create(client, params, opts \\ []) do
    Notifique.request(client, :post, "/email/domains", params, opts)
  end

  def get(client, id, opts \\ []) do
    Notifique.request(client, :get, "/email/domains/#{Notifique.encode_path_segment(id)}", nil, opts)
  end

  def verify(client, id, opts \\ []) do
    Notifique.request(client, :post, "/email/domains/#{Notifique.encode_path_segment(id)}/verify", nil, opts)
  end
end
